api/agent/drivers/driver.go

```diff
@@ -269,17 +269,24 @@ type Config struct {
 }
 
 // https://github.com/fsouza/go-dockerclient/blob/master/misc.go#L166
-func parseRepositoryTag(repoTag string) (repository string, tag string) {
+func parseRepositoryTag(repoTag string) (repository, tag string) {
 	parts := strings.SplitN(repoTag, "@", 2)
+	var digest string
+	if len(parts) == 2 {
```

Comment:
Is there any validation elsewhere that ensures that there's only a single '@' in this string?  E.g. if this function gets more parts than expected, is it safe to proceed?

Reply:
Before we reach here, we would have fetched the registry token in LB. If the image is not valid, we will fail fetching token. But I changed the logic here to use Split and decode digest only if there are two parts